python/grass/script/tests/grass_script_core_location_test.py

```diff
@@ -1,14 +1,15 @@
 """Test functions in grass.script.setup"""
 
 import multiprocessing
+import os
+
+import pytest
 
 import grass.script as gs
 
 
-# All init tests change the global environment, but when it really matters,
-# we use a separate process.
-# Ideally, the functions would support env parameter and the test
-# would mostly use that.
+# This is useful when we want to ensure that function like init does
+# not change the global environment.
 def run_in_subprocess(function):
```

Comment:
Half related, half unrelated:

Related: this seems to be code that could be a fixture, a bit like the example here https://docs.pytest.org/en/8.0.x/how-to/fixtures.html#factories-as-fixtures

Half-Unrelated: if the fixtures, for setup and tear down would be correctly used, it would make this function useless, and would allow to test faster, as we are already using multiple workers for pytest, and calling subproccesses/multiprocessing so much (not only here, but a lot of the python calls do too) adds up somewhere. And [for Windows and by default on macOS](https://pythonforthelab.com/blog/differences-between-multiprocessing-windows-and-linux/), processes are spawned, while on Linux the default is fork, but spawn is available. Spawn is slower, and needs to be able to pickle. So, if we'd want to run these pytest tests on windows one day, we might like that it isn't too long.

Reply:
> if the fixtures, for setup and tear down would be correctly used, it would make this function useless, and would allow to test faster, as we are already using multiple workers for pytest

The idea here is to test functioning in global environment, but to stay isolated at the same time. What is not clear to me is if that will be achieved through the workers of pytest or if it is needed here.

Here, it is just one or two tests, so no big deal either way, but it seems to me that something like this will be needed to test the C functions used through ctypes.

I won't go into this in this PR. Not even the fixture. It is becoming large already.